src/integrator/cuda/cuda_device_properties.hpp

```diff
@@ -10,6 +10,14 @@ static constexpr uint32_t max_warps_per_thread_block =
   max_threads_per_thread_block / warp_size;
 
 static constexpr uint32_t max_submat_blocks = 10;
+
+// Properties for weight algorithm
+static constexpr uint32_t weight_unroll = 4;
+static_assert(weight_unroll == 4, "Weight unroll is only tested for value of 4");
+static constexpr uint32_t weight_thread_block = 640;
```

Comment:
A thread block size of 640 allows for 63% occupancy. There are other sizes which allow for this, however I found this performed the best.

I also found that 4 was the optimal unroll factor through experimentation. A lower unroll factor gives lower register usage and more efficient early exiting, but a higher unroll factor hides the memory latency better.

Reply:
Hmm, ok. I think it should be fine to leave as is for now (noting that this will not generally be portable across systems with a different SM count / max block size, etc). We should probably add a configure file somewhere that sets these for specific hardware (since they are empirically measured)